src/parser/attribute.rs

```diff
@@ -0,0 +1,56 @@
+extern crate xml;
+extern crate case;
+
+use case::CaseExt;
+use parser::util::*;
+
+fn transform_attribute (attribute: String) -> String {
+    let attribute_key_value: Vec<&str> = attribute.split("=").collect();
+    let attribute_key: &str = attribute_key_value[0];
+    let attribute_value: &str = attribute_key_value[1];
+    return formatted_attribute(attribute_key, attribute_value.clone());
+}
+
+fn formatted_attribute ( attribute_key: &str, attribute_value: &str) -> String {
```

Comment:
This area needs work going forward. It's a lot of chaining, that might be hard to follow or debug.